src/control/server/harness.go

```diff
@@ -240,14 +243,27 @@ func (h *IOServerHarness) Start(parent context.Context) error {
 }
 
 // StartManagementService starts the DAOS management service on this node.
+//
+// Bootstrap instance will never join, so explicitly record as first system member
+// if management service gets bootstrapped.
 func (h *IOServerHarness) StartManagementService(ctx context.Context) error {
 	h.RLock()
 	defer h.RUnlock()
 
-	for _, instance := range h.instances {
-		if err := instance.StartManagementService(); err != nil {
+	for _, i := range h.instances {
+		newMember, err := i.StartManagementService()
+		if err != nil {
 			return err
 		}
+		if newMember != nil {
```

Comment:
This is a little confusing. After looking at the changes to StartManagementService(), I guess that newMember can be nil if this method is called on a non-MS instance, right? Generally, using nil as a valid return value isn't desirable because it's atypical and requires a greater level of care in its use.

At a minimum, I'd suggest renaming the variable to something like msReplicaMember to make it clear that it's special, as well as checking for the nil case and returning early (along with a comment explaining this) so that you can outdent the rest of the logic. Even better would probably be to refactor things a bit to avoid calling StartManagementService() unless we're reasonably certain that it should succeed.